plugins/basic/tk_create_menus/generateMenuItems.py

```diff
@@ -54,7 +56,15 @@ class {className}
     "overrideReferences": false,
     "precompiledReferences": [],
     "autoReferenced": true,
-    "defineConstraints": []
+    "defineConstraints": [],
+    "versionDefines": [
+        {{
+            "name": "com.unity.integrations.shotgun",
+            "expression": "",
+            "define": "SHOTGUN_INSTALLED"
+        }}
+    ],
+    "noEngineReferences": false
```

Comment:
what does `noEngineReferences` do?

Reply:
if it's true , it removes all references to UnityEditor and UnityEngine from that assemble file. I don't think we want that.
Either way, json code added to the asmdef file is added based on the link below. It's automatically generated if you want to define something from the editor.
https://docs.unity3d.com/Manual/ScriptCompilationAssemblyDefinitionFiles.html#define-symbols

Reply:
ok, so you did not add this yourself, it was added automatically (I guess the version of Unity you are using adds that now)